### PR TITLE
Fix racing in janitor test

### PR DESCRIPTION
**Pull request by krzyzacy**
fix https://github.com/kubernetes/test-infra/issues/3754

also fixed some wrong set-up :-)

/assign @spxtr

### CODE REVIEW

**Comment by spxtr** [comment]
/lgtm

Although you don't necessarily need to lowercase your log messages XD

File: boskos/janitor/janitor_test.go [hunk 1]
```diff
@@ -101,34 +101,32 @@ func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
...
 	types := []string{"a", "b", "c", "d"}
 	fb := CreateFakeBoskos(1000, types)
 
-	buffer := setup(fb, poolSize, bufferSize)
+	buffer := setup(fb, poolSize, bufferSize, fakeClean)
 	totalAcquire := run(fb, buffer, []string{"t"})
 
 	if totalAcquire != len(fb.resources) {
-		t.Errorf("Expect to acquire all resources(%d) from fake boskos, got %d", len(fb.resources), totalAcquire)
+		t.Errorf("expect to acquire all resources(%d) from fake boskos, got %d", len(fb.resources), totalAcquire)
```
Comment (krzyzacy) [inline boskos/janitor/janitor_test.go]: ah.. right.. doesn't hurt :-)

**Comment by krzyzacy** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/3761/pull-test-infra-bazel/5230/ :-(

**Pull request merged**